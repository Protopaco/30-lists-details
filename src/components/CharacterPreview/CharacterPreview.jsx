import PropTypes from 'prop-types';
import React from 'react';
import styles from './CharacterPreview.css';
import { Link } from 'react-router-dom';

const CharacterPreview = ({ photoUrl, name, id }) => (
    <figure
        className={styles.CharacterPreview}>
        <Link to={`/details/${id}`}>
            <img src={photoUrl} className={styles.previewImage} />
            <figcaption className={styles.previewName}>{name}</figcaption>
        </Link>
    </figure>

);

CharacterPreview.propTypes = {
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
}

export default CharacterPreview;