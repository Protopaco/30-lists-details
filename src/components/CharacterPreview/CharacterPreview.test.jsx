import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterPreview from './CharacterPreview.jsx';

describe('CharacterPreview component', () => {
    afterEach(() => cleanup());
    it('renders CharacterPreview', () => {
        const { asFragment } = render(<Router>
            <CharacterPreview
                photoUrl={'f'}
                name={'f'}
                _id={'f'}
            />
        </Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});
