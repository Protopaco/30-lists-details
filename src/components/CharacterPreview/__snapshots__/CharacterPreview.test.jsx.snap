// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterPreview component renders CharacterPreview 1`] = `
<DocumentFragment>
  <figure
    class="CharacterPreview"
  >
    <a
      href="/details/f"
    >
      <img
        class="previewImage"
        src="f"
      />
      <figcaption
        class="previewName"
      >
        f
      </figcaption>
    </a>
  </figure>
</DocumentFragment>
`;
